backend/src/api/bitcoin/bitcoin.routes.ts

```diff
@@ -241,6 +243,74 @@ class BitcoinRoutes {
     }
   }
 
+  /**
+   * Takes the PSBT as text/plain body, parses it, and adds the full
+   * parent transaction to each input that doesn't already have it.
+   * This is used for BTCPayServer / Trezor users which need access to
+   * the full parent transaction even with segwit inputs.
+   * It will respond with a text/plain PSBT in the same format (hex|base64).
+   */
+  private async postPsbtCompletion(req: Request, res: Response) {
+    res.setHeader('content-type', 'text/plain');
+    const notFoundError = `Couldn't get transaction hex for parent of input`;
+    try {
+      let psbt: bitcoinjs.Psbt;
+      let format: 'hex' | 'base64';
+      let isModified = false;
+      try {
+        psbt = bitcoinjs.Psbt.fromBase64(req.body);
+        format = 'base64';
+      } catch (e1) {
+        try {
+          psbt = bitcoinjs.Psbt.fromHex(req.body);
+          format = 'hex';
+        } catch (e2) {
+          throw new Error(`Unable to parse PSBT`);
+        }
+      }
```

Comment:
Yes, but there are other parsing errors that can occur other than the headers. Adding a header check is superfluous since these check both the headers and everything else.

Reply:
true, it will work this way

(if error messages from bitcoinjs would be sent back to the client this might get confusing. a base64 psbt that can't get parsed as base64 will be parsed as hex, which will obviously fail. which error is sent to the client, `e1` or `e2`?)